src/FileWriterTask.cpp

```diff
@@ -49,13 +50,23 @@ void FileWriterTask::addSource(Source &&Source) {
 void FileWriterTask::InitialiseHdf(std::string const &NexusStructure,
                                    std::vector<StreamHDFInfo> &HdfInfo) {
   auto NexusStructureJson = hdf_parse(NexusStructure, Logger);
+  std::string ErrorString;
+  if (std::filesystem::exists(Filename)){
+    ErrorString = fmt::format("Failed to initialize HDF file \"{}\". Error was: \"{}\"", Filename,
+                              "A file with that filename already exists in that directory. Delete the existing file or provide another filename.");
+    std::throw_with_nested(std::runtime_error(ErrorString));
+  }
+  else if(not FileDirectory.empty() and not std::filesystem::exists(FileDirectory)){
```

Comment:
Why do we need the dir to be empty? Or this checks something else?

Reply:
it should not be empty. Apparently you are not obliged to provide a directory to the file writer (I suppose it just writes it in the bin directory directly then and that will exist).